web/server/codechecker_server/api/report_server.py

```diff
@@ -1299,6 +1305,12 @@ def getDiffResultsHash(self, run_ids, report_hashes, diff_type,
                 if run_ids:
                     results = results.filter(Report.run_id.in_(run_ids))
 
+                if tag_ids:
+                    results = results.outerjoin(
+                        RunHistory, RunHistory.run_id == Report.run_id) \
+                        .filter(RunHistory.id.in_(tag_ids)) \
+                        .filter(get_open_reports_date_filter_query())
+
```

Comment:
This part is duplicated 3 times. It would be good to create a helper function for it and use it all these places:
```py
                if run_ids:
                    results = results.filter(Report.run_id.in_(run_ids))

                if tag_ids:
                    results = results.outerjoin(
                        RunHistory, RunHistory.run_id == Report.run_id) \
                        .filter(RunHistory.id.in_(tag_ids)) \
                        .filter(get_open_reports_date_filter_query())

```